src/renderer/components/kube-object-menu/inject-naive/inject-naive.tsx

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import React from "react";
+import { editResourceTab } from "../../dock/edit-resource.store";
+import { hideDetails } from "../../kube-detail-params";
+import { apiManager } from "../../../../common/k8s-api/api-manager";
+import { KubeObjectMenuRegistry } from "../../../../extensions/registries";
+import { catalogEntityRegistry } from "../../../api/catalog-entity-registry";
+
+
+interface Props {
+  Component: React.ReactType<any>
+  [key: string]: any,
```

Comment:
Also I really don't line this lack of type safety

Reply:
Well yeah, of course. But as said, `InjectNaive` is just hard-coded solution which is not ready to be merged. :) It's here just to pass the dependencies to `KubeObjectMenu` and therefore not scalable.

Reply:
Type safety is now in place: #4338.